File: 4th semester/SDI (Systems for Design & Implementation)/Labs/Lab. 12/bookstore-web/src/main/webapp/src/app/books/shared/book.service.ts
import {Injectable} from '@angular/core';

import {HttpClient} from "@angular/common/http";

import {Observable} from "rxjs";
import {Book} from "./book.model";
import {map} from "rxjs/operators";


@Injectable()
export class BookService {
  private booksUrl = 'http://localhost:8080/api/books';

  constructor(private httpClient: HttpClient) {
  }

  getBooks(): Observable<Book[]> {
    return this.httpClient
      .get<Array<Book>>(this.booksUrl);
  }

  getBook(id: number): Observable<Book> {
    console.log(id);
    return this.getBooks()
      .pipe(
        map(books => books.find(book => book.id === id))
      );
  }

  saveBook(book: Book): Observable<Book> {
    console.log("saveBook", book);

    return this.httpClient
      .post<Book>(this.booksUrl, book);
  }

  updateBook(book): Observable<Book> {
    const url = `${this.booksUrl}/${book.id}`;
    return this.httpClient
      .put<Book>(url, book);
  }

  deleteBook(id: number): Observable<any> {
    const url = `${this.booksUrl}/${id}`;
    return this.httpClient
      .delete(url);
  }
}

File: IV/SDI (Systems for Design & Implementation)/Labs/Lab. 12/bookstore-core/src/main/java/ro/ubb/bookstore/core/service/BookService.java
package ro.ubb.bookstore.core.service;

import ro.ubb.bookstore.core.model.Book;
import ro.ubb.bookstore.core.model.CustomerBook;

import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.Set;

public interface BookService {
    Book getBook(Long bookId);
    List<Book> getAllBooks();
    Book saveBook(Book book);
    Book updateBook(Book book);
    void deleteBook(Long bookId);
}
